backend/irc/ircconnection.ts

```diff
@@ -85,6 +86,24 @@ rl.on('line', async (line) => {
   } else if (ircMessage.command == 'PRIVMSG' && !ircMessage.prefix?.toLowerCase().includes('bot@')) {
     if (Date.now() - joinConfig.bufferTime.getTime() < 5000) {
       return;
+    } else if (ircMessage.command == 'PRIVMSG') {
+      const msg = ircMessage.params.slice(1).join(' ');
+      if (msg.match(/Duck friend scores in #/i)) {
+        const splitMsgByBullet = msg.split('\u2022');
+        // fix the first split by removing the 'Duck ... scores in #channel'
+        const correctFirstScore = splitMsgByBullet[0].split(':');
+        splitMsgByBullet[0] = `${correctFirstScore[2]}: ${correctFirstScore[3]}`;
+        await insertOrUpdateFriendScores(splitMsgByBullet);
+        myEmitter.emit('friendScore', splitMsgByBullet);
+        await DatabaseDuccUtils.retrieveAllDuccScores();
+      } else if (msg.match(/Duck killer scores in #/i)) {
+        const splitMsgByBullet: string[] = msg.split('\u2022');
+        // fix the first split by removing the 'Duck ... scores in #channel'
+        const correctFirstScore: string[] = splitMsgByBullet[0].split(':');
+        splitMsgByBullet[0] = `${correctFirstScore[2]}: ${correctFirstScore[3]}`;
+        await insertOrUpdateKillerScores(splitMsgByBullet);
+        myEmitter.emit('killedScore', splitMsgByBullet);
+      }
```

Comment:
These are so similar they could be made into one function with a friend/killer argument.